### ISSUE TITLE
Name parsing issue

### ISSUE THREAD

**Issue by brettz9** [syavorsky/comment-parser#112]
Hi,

As per https://github.com/gajus/eslint-plugin-jsdoc/issues/673 , we're also getting an issue with name parsing, e.g., this simplified example triggers `spec:name:invalid-default` (i.e., 'invalid default value syntax'):

```js
      /**
       * @param {Function} [processor=data => data] A function to run
       */
      function processData(processor) {
        return processor(data)
      }
```

I'm not sure there should be such an "invalid default" rule given that equal signs may be part of default functions (though the separate empty default issue still stands).

**Comment by syavorsky** [comment]
Will take care of this

**Comment by syavorsky** [comment]
fixed in `1.1.0`, please see [tests/e2e/issue-112](https://github.com/syavorsky/comment-parser/blob/master/tests/e2e/issue-112.spec.js)

**Comment by brettz9** [comment]
Appears to be working like a charm, thanks much as usual! (FWIW, released on our end in `eslint-plugin-jsdoc` v31.0.6).

In case you weren't aware, your 1.1.0 was the last I checked still marked as a beta.

**Comment by syavorsky** [comment]
Great. Will drop the beta tag
